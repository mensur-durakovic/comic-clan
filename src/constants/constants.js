// grouping categories
export const YEAR = "Year";
export const WRITER = "Writer";
export const ARTIST = "Artist";
export const OWNER = "Owner";
export const RANDOM = "Random";
export const GROUPING_CATEGORIES = [YEAR, WRITER, ARTIST, OWNER, RANDOM];

// website texts
export const SEARCH_PLACEHOLDER = "Search by book name";
export const BACK_TEXT = "Back to collection";
export const OWNED_BY = "Owned By";
export const PUBLICATION = "Publication";
export const LOADING_TEXT = "Loading";

// api related stuff
export const COMMIC_CLAN_BASE_URL = "https://comicclan.vett.io/comics";

// text messages
export const NO_RESULTS_TEXT = "Ooops, looks like that comic doesn't exist! :(";
export const RANDOM_BOOKS_TITLE = "Other random books";

// error messages
export const ERROR_FETCH_BOOKS_TEXT = "Ooops, an error occurred!";
export const ERROR_SELECT_BOOK_TEXT = "Ooops, looks like that comic doesn't exist! Feel free to take a look at comics below :)";
export const ERROR_404_TEXT = "404 - page does not exist!";