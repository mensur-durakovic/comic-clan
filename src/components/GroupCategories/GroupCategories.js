import React from "react";
import styled from "styled-components";
import GroupCategory from "./GroupCategory/GroupCategory";
import { GROUPING_CATEGORIES } from "../../constants/constants";

const GroupCategoriesWrapper = styled.div`
  padding-left: 30px;
  padding: 0 30px;
  display: flex;
  flex-wrap: wrap;
`;

const GroupCategories = React.memo(() => {
  return (
    <GroupCategoriesWrapper>
      {GROUPING_CATEGORIES.map(c => (
        <GroupCategory
          key={c}
          categoryName={c}
        />
      ))}
    </GroupCategoriesWrapper>
  );
});

export default GroupCategories;
