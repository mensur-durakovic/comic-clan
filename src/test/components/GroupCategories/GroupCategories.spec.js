import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { YEAR } from "../../../constants/constants";
import { commicBooks, groupedBooks } from "../../testUtilities/constants";
import GroupCategories from "../../../components/GroupCategories/GroupCategories";

const mockStore = configureStore([]);

describe("GroupCategories component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      books: commicBooks,
      groupedBooks: groupedBooks,
      activeCategory: YEAR
    });
    component = renderer.create(
      <Provider store={store}>
        <GroupCategories/>
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  
});
