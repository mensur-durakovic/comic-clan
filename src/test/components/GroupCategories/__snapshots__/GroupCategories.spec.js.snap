// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupCategories component should render with given state from Redux store 1`] = `
<div
  className="sc-AxirZ gOmwfh"
>
  <div
    className="sc-AxjAm gUnyex"
    onClick={[Function]}
  >
    Year
  </div>
  <div
    className="sc-AxjAm hemMAk"
    onClick={[Function]}
  >
    Writer
  </div>
  <div
    className="sc-AxjAm hemMAk"
    onClick={[Function]}
  >
    Artist
  </div>
  <div
    className="sc-AxjAm hemMAk"
    onClick={[Function]}
  >
    Owner
  </div>
  <div
    className="sc-AxjAm hemMAk"
    onClick={[Function]}
  >
    Random
  </div>
</div>
`;
